### PR TITLE
Improve code quality

### PR DESCRIPTION
**Pull request by AlWo223**

### CODE REVIEW

**Review by skylee03 — APPROVED** [review]
LGTM!

**Comment by AlWo223** [comment]
I have done some major refactoring and optimization in ActivityParser as well, would be great if you can review these changes as well

**Review by nihalzp — APPROVED** [review]
LGTM!

File: src/main/java/athleticli/parser/ActivityParser.java [hunk 1]
```diff
@@ -20,45 +19,87 @@
...
     //@@author  AlWo223
     /**
-     * Parses the index of an activity.
+     * Parses the index of an activity from a string input.
      *
-     * @param commandArgs The raw user input containing the index.
-     * @return index                The parsed Integer index.
-     * @throws AthletiException If the input is not an integer.
+     * @param commandArgs       Raw user input containing the index.
+     * @return index            The parsed Integer index.
```
Comment (skylee03) [inline src/main/java/athleticli/parser/ActivityParser.java:25]:
```suggestion
     * @return                  The parsed Integer index.
```

`@return` field does not need to have a name like `@param`.

File: src/main/java/athleticli/parser/ActivityParser.java [hunk 2]
```diff
@@ -20,45 +19,87 @@
...
+     */
+    public static ActivityChanges parseSwimEdit(String arguments) throws AthletiException {
+        return parseActivityTypeEdit(arguments, false);
     }
 
     /**
      * Parses the provided swim arguments of the edit command.
+     *
      * @param arguments         The raw user input containing the updated swim.
      * @return activityChanges  The parsed ActivityChanges object.
```
Comment (skylee03) [inline src/main/java/athleticli/parser/ActivityParser.java:104]:
```suggestion
     * @return                  The parsed ActivityChanges object.
```

File: src/main/java/athleticli/parser/ActivityParser.java [hunk 3]
```diff
@@ -76,6 +117,7 @@ public static ActivityChanges parseSwimChanges(String arguments) throws AthletiE
 
     /**
      * Parses the provided run or cycle arguments of the edit command.
+     *
      * @param arguments         The raw user input containing the updated run or cycle.
      * @return activityChanges  The parsed ActivityChanges object.
```
Comment (skylee03) [inline src/main/java/athleticli/parser/ActivityParser.java:122]:
```suggestion
     * @return                  The parsed ActivityChanges object.
```

File: src/main/java/athleticli/parser/ActivityParser.java [hunk 4]
```diff
@@ -93,6 +135,7 @@ public static ActivityChanges parseRunCycleChanges(String arguments) throws Athl
 
     /**
      * Parses the provided activity arguments of the edit command.
+     *
      * @param arguments         The raw user input containing the updated activity.
      * @return activityChanges  The parsed ActivityChanges object.
```
Comment (skylee03) [inline src/main/java/athleticli/parser/ActivityParser.java:140]:
```suggestion
     * @return                  The parsed ActivityChanges object.
```

File: src/main/java/athleticli/parser/ActivityParser.java [hunk 5]
```diff
@@ -108,46 +151,67 @@ public static ActivityChanges parseActivityChanges(String arguments) throws Athl
...
     }
 
+    /**
+     * Finds the index of the next separator in the arguments String
+     *
+     * @param arguments Raw user input containing the arguments.
+     * @param startIndex The String position index to start searching from.
+     * @param separators List of separators to search for.
+     * @param currentSeparatorIndex Index of the current separator, refers to the list of separators.
+     * @return endIndex The String position index of the next separator.
```
Comment (skylee03) [inline src/main/java/athleticli/parser/ActivityParser.java:196]:
```suggestion
     * @return          The String position index of the next separator.
```

File: src/main/java/athleticli/parser/ActivityParser.java [hunk 6]
```diff
@@ -186,100 +250,30 @@ public static void parseSegment(ActivityChanges activityChanges, String segment,
...
-    }
-
-    /**
-     * Parses the index of an activity update for the edit command.
+     * Parses the index of an activity update for the edit command from the provided arguments.
      *
      * @param arguments The raw user input containing the index.
-     * @return index              The parsed Integer index.
-     * @throws AthletiException If the input format is invalid
+     * @return index The parsed Integer index.
```
Comment (skylee03) [inline src/main/java/athleticli/parser/ActivityParser.java:256]:
```suggestion
     * @return       The parsed Integer index.
```

File: src/main/java/athleticli/parser/ActivityParser.java [hunk 7]
```diff
@@ -186,100 +250,30 @@ public static void parseSegment(ActivityChanges activityChanges, String segment,
...
     }
 
     /**
-     * Parses the raw user input for viewing the activity list and returns whether the user wants the detailed
-     * view
+     * Parses the raw user input for viewing the activity list and returns whether the user wants the detailed view
      *
      * @param commandArgs The raw user input containing the arguments.
-     * @return boolean      Whether the user wants the detailed view.
+     * @return boolean Whether the user wants the detailed view.
```
Comment (skylee03) [inline src/main/java/athleticli/parser/ActivityParser.java:271]:
```suggestion
     * @return         Whether the user wants the detailed view.
```

**Review by skylee03 — APPROVED** [review]
LGTM!
`@return` field does not need to have a name like `@param`.

**Pull request merged**